CoreScriptsRoot/ServerCoreScripts/ClientChat/ChatBar.lua

```diff
@@ -0,0 +1,246 @@
+local source = [[
+local module = {}
+--////////////////////////////// Include
+--//////////////////////////////////////
+local modulesFolder = script.Parent
+local moduleTransparencyTweener = require(modulesFolder:WaitForChild("TransparencyTweener"))
+local moduleChatSettings = require(modulesFolder:WaitForChild("ChatSettings"))
+local ClassMaker = require(modulesFolder:WaitForChild("ClassMaker"))
+
+--////////////////////////////// Methods
+--//////////////////////////////////////
+local methods = {}
+
+local function CreateGuiObject()
+	local backgroundImagePixelOffset = 8
+	local textBoxPixelOffset = 8
+
+	local BaseFrame = Instance.new("Frame")
+	BaseFrame.Size = UDim2.new(1, 0, 1, 0)
+	BaseFrame.BackgroundTransparency = 0.6
+	BaseFrame.BorderSizePixel = 0
+	BaseFrame.BackgroundColor3 = Color3.new(0, 0, 0)
+
+	local BoxFrame = Instance.new("Frame", BaseFrame)
+	BoxFrame.Name = "BoxFrame"
+	BoxFrame.BackgroundTransparency = 0.6
+	BoxFrame.BorderSizePixel = 0
+	BoxFrame.BackgroundColor3 = Color3.new(1, 1, 1)
+	BoxFrame.Size = UDim2.new(1, -backgroundImagePixelOffset * 2, 1, -backgroundImagePixelOffset * 2)
+	BoxFrame.Position = UDim2.new(0, backgroundImagePixelOffset, 0, backgroundImagePixelOffset)
+
+	local TextBox = Instance.new("TextBox", BoxFrame)
+	TextBox.Name = "ChatBar"
+	TextBox.BackgroundTransparency = 1
+	TextBox.Size = UDim2.new(1, -textBoxPixelOffset * 2, 1, -textBoxPixelOffset * 2)
+	TextBox.Position = UDim2.new(0, textBoxPixelOffset, 0, textBoxPixelOffset)
+	TextBox.FontSize = moduleChatSettings.ChatBarTextSize
+	TextBox.Font = Enum.Font.SourceSansBold
+	TextBox.TextColor3 = Color3.new(1, 1, 1)
+	TextBox.TextStrokeTransparency = 0.75
+	TextBox.ClearTextOnFocus = false
+	TextBox.TextXAlignment = Enum.TextXAlignment.Left
+	TextBox.TextYAlignment = Enum.TextYAlignment.Top
+	TextBox.TextWrapped = true
+	TextBox.Text = ""
+
+	local TextLabel = Instance.new("TextLabel", BoxFrame)
+	TextLabel.BackgroundTransparency = 1
+	TextLabel.Size = TextBox.Size
+	TextLabel.Position = TextBox.Position
+	TextLabel.FontSize = TextBox.FontSize
+	TextLabel.Font = TextBox.Font
+	TextLabel.TextColor3 = TextBox.TextColor3
+	TextLabel.TextStrokeTransparency = TextBox.TextStrokeTransparency
+	TextLabel.TextXAlignment = TextBox.TextXAlignment
+	TextLabel.TextYAlignment = TextBox.TextYAlignment
+	TextLabel.Text = "This value was un-initialized lololololololololol"
```

Comment:
The idea was that a function would be called later to set the default text based on if it was mobile or PC. This way you could make it say "click to chat" or "touch to chat".